src/wallet/db.h

```diff
@@ -87,6 +87,33 @@ class CDBEnv
 
 extern CDBEnv bitdb;
 
+/** An instance of this class represents one database.
+ * For BerkeleyDB this is just a (env, strWalletFile) tuple.
+ **/
+class CWalletDBWrapper
+{
+public:
+    CWalletDBWrapper(CDBEnv *env_in, const std::string &strWalletFile_in):
+        env(env_in), strWalletFile(strWalletFile_in)
+    {
+    }
+
+    /** Rewrite the entire database on disk, with the exception of key pszSkip if non-zero
+     */
+    bool Rewrite(const char* pszSkip=nullptr);
+
+    /** Back up the entire database to a file.
+     */
+    bool Backup(const std::string& strDest);
+
+    /** Get a name for this database, for debugging etc.
+     */
+    std::string GetName() { return strWalletFile; }
```

Comment:
They are subtly different, only happen to be the same for BerkeleyDB: 

- `GetName` is an an abstract name to refer to the database object for logging. It should exist for any version of CDBWrapper.
- `strWalletFile` is an implementation detail that only exists for BerkeleyDB. Other databases may have completely different connection descriptions, which do not involve simply a string. So this is private to the low-level implementation (CDB), the higher-level stuff (such as CWallet) should not refer to it.